bundles/org.eclipse.swt/Eclipse SWT Custom Widgets/common/org/eclipse/swt/custom/BusyIndicator.java

```diff
@@ -70,6 +70,41 @@ public static void showWhile(Display display, Runnable runnable) {
 		}
 	}
 
+	/**
+	 * If called from a {@link Display} thread, waits for the given
+	 * <code>Future</code> to complete and provides busy feedback using the
+	 * busy indicator. While waiting for completion, pending UI events are processed
+	 * to prevent UI freeze.
+	 *
+	 * If there is no {@link Display} for the current thread, the method returns
+	 * immediately and no feedback is shown to the user.
+	 *
+	 * In any case, the passed {@link Future} is returned for chaining of further operations.
+	 *
+	 * @param future the {@link Future} for which busy feedback is to be shown.
+	 * @return the {@link Future} for chaining.
+	 * @since 3.127
+	 */
+	public static <F extends Future<T>, T> F showWhile(F future) {
+		Objects.requireNonNull(future);
+		if (!future.isDone()) {
+			Display display = Display.findDisplay(Thread.currentThread());
+			if (display != null) {
+				Integer busyId = setBusyCursor(display);
```

Comment:
Better use existing showWhile(monitor, Runnable) here - that way this static method does not depend on internals.

Reply:
This would perform all checks I have done before again + requires creating an additional runnable wrapper.